lib/api/model/model.dart

```diff
@@ -635,6 +645,19 @@ class DmConversation extends Conversation {
 
   DmConversation({required this.allRecipientIds})
     : assert(isSortedWithoutDuplicates(allRecipientIds.toList()));
+
+  @override
+  bool isSameAs(Conversation other) {
+    if (other is! DmConversation) return false;
+    final xs = allRecipientIds;
+    final ys = other.allRecipientIds;
+    if (xs.length != ys.length) return false;
+    final xs_ = xs.iterator; final ys_ = ys.iterator;
+    while (xs_.moveNext() && ys_.moveNext()) {
+      if (xs_.current != ys_.current) return false;
+    }
+    return true;
```

Comment:
This is best kept as its own method with its own name. It's written to a low-level API, but has a meaning that can be described very crisply and abstractly.